spacecmd/src/spacecmd/system.py

```diff
@@ -4238,3 +4238,123 @@ def do_system_scheduleapplyconfigchannels(self, args):
     return 0
 
 ####################
+
+
+def help_system_listmigrationtargets(self):
+    print('system_listmigrationtargets: List possible migration targets for given systems.')
+    print('usage: system_listmigrationtargets <SYSTEMS>')
+    print('')
+    print(self.HELP_SYSTEM_OPTS)
+
+
+def do_system_listmigrationtargets(self, args):
+    arg_parser = get_argument_parser()
+
+    (args, options) = parse_command_arguments(args, arg_parser)
+
+    if len(args) != 1:
+        self.help_system_listmigrationtargets()
+        return
+
+    # use the systems listed in the SSM
+    if re.match('ssm', args[0], re.I):
+        systems = self.ssm.keys()
+    else:
+        systems = self.expand_systems(args[0])
+
+    if not systems:
+        print('No systems found')
+        return
+
+    for system in sorted(systems):
+        system_id = self.get_system_id(system)
+        if not system_id:
+            print('WARN: Cannot find system ' + str(system))
+            continue
+
+        print('System ' + str(system))
+        tgts = self.client.system.listMigrationTargets(self.session, system_id)
+
+        if not tgts:
+            print('  No migration targets')
+
+        for num, tgt in enumerate(tgts, start=1):
+            print('  Target #' + str(num) + ":")
+            print('    IDs: ' + tgt['ident'])
+            print('    Friendly names: ' + tgt['friendly'])
+
+####################
+
+
+def help_system_schedulespmigration(self):
+    print('system_schedulespmigration: Schedule a Service Pack migration for systems.')
+    print('usage: system_listmigrationtargets <SYSTEM> <DRY_RUN?> <BASE_CHANNEL_LABEL> [options] \
+\n    options: \
+\n        -s START_TIME \
+\n        -t MIGRATION_TARGET_ID (see system_listmigrationtargets) \
+\n        -c CHILD_CHANNELS (comma-separated child channels labels)')
```

Comment:
why is migration target a option and dryrun an argument?
It is not easy to understand how to provide it. Especially as dryrun is a bool and it is not explained how to provide it.

What about to turn it around? Migration Target as mandatory argument and dryrun as optional option?

Reply:
In the Uyuni API, the dry run is mandatory parameter, and migration target is optional (there are 2 `scheduleSPMigration` methods: one accepts the `targetIdent`, one does not). So I think that migration target should stay optional.

About dry run: I could make it optional as well (maybe using a `-d` flag?). When user does not provide it, it would default to `false`. What do you think?